packages/discovery-provider/src/api/v1/helpers.py

```diff
@@ -351,9 +362,144 @@ def extend_track(track, session=None):
             track["download_conditions"], session
         )
 
+    track["preview"] = get_preview_url_with_mirrors(track, current_user_id)
+    track["stream"] = {"url": None, "mirrors": []}
+    track["download"] = {"url": None, "mirrors": []}
+    if track.get("access", False):
+        authed_user_id = recover_auth_user_id_from_signature_headers()
```

Comment:
Unsure how best to prevent repeat calls of this without further inflating the params to extend_track and touching every call site... though maybe I should touch every call site anyway to add the headers to the parsers?

Also haven't addressed the 3rd party API case yet